### PR TITLE
Support ProtoDataSotre

### PR DESCRIPTION
**Pull request by S-H-Y-A**
To konw how to use, look updated README.md and sample app.

### CODE REVIEW

**Comment by alorma** [comment]
Hi @S-H-Y-A

Thanks for this PR, it looks promising!!

I will take a look this weekend and give you feedback!

File: app/src/main/java/com/alorma/compose/settings/example/ui/screens/CheckboxesScreen.kt [hunk 1]
```diff
@@ -71,24 +76,50 @@ fun CheckboxesScreen(navController: NavHostController) {
...
+          key = "PreferenceDataStore",
+          state = preferenceDataStoreStorage,
+        )
+      },
+    )
+    Divider()
+    val dataStoreState = rememberProtoDataStoreState(
+      serializer = SettingsSerializer
+    )
+    val protoDataStoreStorage = rememberProtoDataStoreTransformSettingState(
```
[the remark this answers is not in the document]
Comment (S-H-Y-A) [reply app/src/main/java/com/alorma/compose/settings/example/ui/screens/CheckboxesScreen.kt:104]: https://github.com/alorma/Compose-Settings/blob/a59962d54388a6903cc89dad9d46663f82b127a3/app/src/main/java/com/alorma/compose/settings/example/ui/screens/CheckboxesScreen.kt#L101-L103

`rememberProtoDataStoreState` just instantiate the datastore.

https://github.com/alorma/Compose-Settings/blob/a59962d54388a6903cc89dad9d46663f82b127a3/app/src/main/java/com/alorma/compose/settings/example/ui/screens/CheckboxesScreen.kt#L104-L108

`encoder` and `decoder` is convert between ProtoDataStore Definition Kt object and the settings value.

[the remark this answers is not in the document]
Comment (S-H-Y-A) [reply app/src/main/java/com/alorma/compose/settings/example/ui/screens/CheckboxesScreen.kt:104]: `decoder` is used as [this](https://developer.android.com/topic/libraries/architecture/datastore#preferences-read).

`encoder` is used as [this](https://developer.android.com/topic/libraries/architecture/datastore#preferences-write).

[the remark this answers is not in the document]
Comment (S-H-Y-A) [reply app/src/main/java/com/alorma/compose/settings/example/ui/screens/CheckboxesScreen.kt:102]: >[ What if we have multiple SettingsTiles that uses proto?](https://github.com/alorma/Compose-Settings/pull/78#discussion_r1044655680)

Call `rememberProtoDataStoreState()` just once and call `rememberProtoDataStoreTransformSettingState()` individually.

```
// User.proto


message User {
  int32 userId = 1;
  string userName = 2;
}
```

```
// Compsable method


 val dataStoreState = rememberProtoDataStoreState( 
   serializer = UserSerializer 
 ) 

// userId
 val protoDataStoreStorage = rememberProtoDataStoreTransformSettingState( 
   protoDataStoreState = dataStoreState, 
   encoder = {proto, newLoginUserId-> proto.toBuilder().setUserId(newLoginUserId).build() }, 
   decoder = { it.userId }, 
 ) 

// userName
 val protoDataStoreStorage = rememberProtoDataStoreTransformSettingState( 
   protoDataStoreState = dataStoreState, 
   encoder = {proto, newLoginUserName -> proto.toBuilder().setUserName(newLoginUserName).build() }, 
   decoder = { it.userName}, 
 ) 
```

[the remark this answers is not in the document]
Comment (S-H-Y-A) [reply app/src/main/java/com/alorma/compose/settings/example/ui/screens/CheckboxesScreen.kt:102]: Otherwize:

Way 1
```
data class UserInfo(
    val id: Int,
    val name: String,
)
```

```
val dataStoreState = rememberProtoDataStoreState( 
    serializer = UserSerializer 
) 

val userData = rememberProtoDataStoreTransformSettingState(
    protoDataStoreState = dataStoreState,
    encoder = {proto, newLoginUser->
        proto
            .toBuilder()
            .setUserId(newLoginUser.id)
            .setUserName(newLoginUser.userName)
            .build()
    },
    decoder = { UserInfo(it.userId, it,userName ) }, 
) 
```

Way 2

```
val dataStoreState = rememberProtoDataStoreState( 
    serializer = UserSerializer 
) 

val userData = rememberProtoDataStoreSettingState(
    protoDataStoreState = dataStoreState
)

var id by remember { mutableStateOf(1) }
var userName by remember { mutableStateOf("S-H-Y-A") }

Button(
    onclick = {
        userData = userData
            .toBuilder()
            .setUserId(id)
            .setUserName(userName)
            .build()
    }
) {
    ...
}
```

[the remark this answers is not in the document]
Comment (S-H-Y-A) [reply app/src/main/java/com/alorma/compose/settings/example/ui/screens/CheckboxesScreen.kt:103]: I addded a sample to the app when we have multiple SettingsTiles with a single protobuf. (ProtoScreen)
@alorma Check, please.

**Review by alorma — APPROVED** [review]

**Comment by alorma** [comment]
I will get a release later this afternoon

**Pull request merged**